llarp/config/config.cpp

```diff
@@ -932,50 +931,97 @@ namespace llarp
             "This is typically required only when incoming= is used to listen on an internal",
             "private range IP address/port that received traffic forwarded from the public IP.",
         },
-        [this](uint16_t arg) { PublicPort = net::port_t::from_host(arg); });
+        [this](uint16_t arg) { public_port = net::port_t::from_host(arg); });
+
+    // DISCUSS: does this need to be an optional?
+    auto parse_addr_for_link = [net_ptr](const std::string& arg) {
+      std::optional<oxen::quic::Address> maybe = std::nullopt;
+      std::string_view arg_v;
 
-    auto parse_addr_for_link = [net_ptr](
-                                   const std::string& arg, net::port_t default_port, bool inbound) {
-      std::optional<SockAddr> addr = std::nullopt;
       // explicitly provided value
       if (not arg.empty())
       {
-        if (arg[0] == ':')
-        {
-          // port only case
-          default_port = net::port_t::from_string(arg.substr(1));
-          if (!inbound)
-            addr = net_ptr->WildcardWithPort(default_port);
-        }
-        else
-        {
-          addr = SockAddr{arg};
-          if (net_ptr->IsLoopbackAddress(addr->getIP()))
-            throw std::invalid_argument{fmt::format("{} is a loopback address", arg)};
-        }
+        arg_v = std::string_view{arg};
       }
-      if (not addr)
+
+      if (arg_v[0] == ':')
       {
-        // infer public address
-        if (auto maybe_ifname = net_ptr->GetBestNetIF())
-          addr = net_ptr->GetInterfaceAddr(*maybe_ifname);
+        uint16_t res = std::atoi(arg_v.substr(1).data());
+
+        maybe = oxen::quic::Address{""s, res ? res : DEFAULT_LISTEN_PORT};
       }
+      else if (auto pos = arg_v.find(':'); pos != arg_v.npos)
+      {
+        auto h = arg_v.substr(0, pos);
+        uint16_t p = std::atoi(arg_v.substr(pos + 1).data());
 
-      if (addr)
+        maybe = oxen::quic::Address{std::string{h}, p};
+
+        // TODO: unfuck llarp/net
+        // if (net_ptr->IsLoopbackAddress(addr->port()))
+        //   throw std::invalid_argument{fmt::format("{} is a loopback address", arg)};
```

Comment:
We should still do this, but via: https://github.com/oxen-io/oxen-libquic/pull/65

Reply:
fixed with your previous commits